Content/GUI/UIDraggablePanel.cs

```diff
@@ -0,0 +1,211 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Terraria.Audio;
+using Terraria.GameContent.UI.Elements;
+using Terraria.ID;
+using Terraria.Localization;
+using Terraria.UI;
+
+namespace PathOfTerraria.Content.GUI;
+
+public class UIDraggablePanel : UIPanel
+{
+	// Stores the offset from the top left of the UIPanel while dragging.
+	private Vector2 Offset { get; set; }
+
+	private bool Dragging { get; set; }
+
+	private readonly bool _stopItemUse;
+
+	private int _uiDelay = -1;
+
+	private readonly UIPanel _header;
+
+	private readonly Dictionary<string, UIPanelTab> _menus;
+
+	public UIDraggablePanel(bool stopItemUse, bool showCloseButton,
+		IEnumerable<(string key, LocalizedText text)> menuOptions)
+	{
+		_stopItemUse = stopItemUse;
+
+		SetPadding(0);
+
+		_header = new UIPanel();
+		_header.SetPadding(0);
+		_header.Width.Set(0, 1f);
+		_header.Height.Set(32, 0f);
+		_header.BackgroundColor.A = 255;
+		_header.OnLeftMouseDown += Header_MouseDown;
+		_header.OnLeftMouseUp += Header_MouseUp;
+		Append(_header);
+
+		if (showCloseButton)
+		{
+			var closeButton = new UITextPanel<char>('X');
+			closeButton.SetPadding(7);
+			closeButton.Width.Set(40, 0);
+			closeButton.Left.Set(-40, 1);
+			closeButton.BackgroundColor.A = 255;
+			_header.Append(closeButton);
+		}
+
+		UIPanel viewArea = new();
+		viewArea.Top.Set(38, 0);
+		viewArea.Width.Set(0, 1f);
+		viewArea.Height.Set(-38, 1f);
+		viewArea.BackgroundColor = Color.Transparent;
+		viewArea.BorderColor = Color.Transparent;
+		Append(viewArea);
+
+		_menus = new Dictionary<string, UIPanelTab>();
+
+		float left = 0;
+
+		foreach ((string key, LocalizedText text) in menuOptions)
+		{
+			UIPanelTab menu;
+			_menus.Add(key, menu = new UIPanelTab(key, text));
+			menu.SetPadding(7);
+			menu.Left.Set(left, 0f);
+			menu.BackgroundColor.A = 255;
+			menu.Recalculate();
+			menu.UseImmediateMode = true;
```

Comment:
This is what fixed the inner panel background. I am not entirely sure why

Reply:
As in, fixed the child elements bleeding from outside of the bounds of the parent element?